mxnet-bot/LabelBotPredict/elastic_beanstalk_server/SentenceParser.py

```diff
@@ -0,0 +1,145 @@
+        r'(?:\S)'                                                                       # anything else
+    ]
+
+    def __init__(self, loggingLevel = 20):
+        self.data = None
+        # extract words stem
+        self.porter = nltk.PorterStemmer()
+        # a set of stopwords
+        self.stops = set(stopwords)
+        logging.basicConfig(level=loggingLevel)
+        pass
+
+    # pandas read csv/json/xlsx files to dataframe format
+    def read_file(self, filepath, filetype, encod='ISO-8859-1', header=None):
+        logger.info('Start reading File')
+        if not os.path.isfile(filepath):
+            logger.error("File Not Exist!")
+            sys.exit()
+        if filetype == 'csv':
+            df = pd.read_csv(filepath, encoding=encod, header=header)
+        elif filetype == 'json':
+            df = pd.read_json(filepath, encoding=encod, lines=False)
+        elif filetype == 'xlsx':
+            df = pd.read_excel(filepath, encoding=encod, header=header)
+        else:
+            logger.error("Extension Type not Accepted!")
+            sys.exit()
+
+        logger.debug(df)
+        self.data = df
+
+    # Operation on the DataFrame
+    def merge_column(self, columns, name):
+        logger.info('Merge headers %s to %s', str(columns), name)
+        self.data[name] = ''
+        for header in columns:
+            self.data[name] += ' ' + self.data[header]
+
+    def get_all_headers(self):
+        return list(self.data.columns.values)
+
+    def get_column(self, column):
+        return self.data[column].values.tolist()
+    
+    def clean_body(self, column, remove_template=True, remove_code=True):
+        # clean issue's description from template
+        logger.info("Start Removing Templates..")
+        for i in range(len(self.data)):
+            # remove 'Environment info' part
+            if remove_template and "## Environment info" in self.data[column][i]:
+                index = self.data.loc[i, column].find("## Environment info")
+                self.data.loc[i, column] = self.data.loc[i, column][:index]
+            # remove code
+            if remove_code and "```" in self.data[column][i]:
+                sample = self.data[column][i].split("```")
+                sample = [sample[i*2] for i in range(0, int((len(sample)+1)/2))]
+                self.data.loc[i,column] = " ".join(sample)
+
+    # Start cleaning the text from column
+    def process_text(self, column, remove_symbol=True, remove_stopwords=False, stemming=False):
+        logger.info("Start Data Cleaning...")
+        # remove some symbols
+        self.data[column] = self.data[column].str.replace(r'[\n\r\t]+', ' ')
+        # remove URLs
+        self.data[column] = self.data[column].str.replace(self.regex_str[3], ' ')
+        tempcol = self.data[column].values.tolist()
+        for i in range(len(tempcol)):
+            row = BeautifulSoup(tempcol[i], 'html.parser').get_text().lower()
```

Comment:
Make sure you add all external dependencies in requirements

Reply:
yes, all external dependencies in the requirements.txt